Tests/AblyChatTests/Mocks/MockRealtimeChannel.swift

```diff
@@ -8,6 +8,11 @@ final class MockRealtimeChannel: NSObject, RealtimeChannelProtocol {
 
     var properties: ARTChannelProperties { .init(attachSerial: attachSerial, channelSerial: channelSerial) }
 
+    // I don't see why the nonisolated(unsafe) keyword would cause a problem when used for tests in this context.
```

Comment:
What if you call `publish()` on one thread and try to read the `lastPublished*` variables on another?

Reply:
let's fix it if it does cause an issue... tests seem fine so far. I don't want to hold this PR up any longer

Reply:
My opinion is that it's going to be confusing to somebody who sees this in the future and thinks that you knew something they didn’t about why this is OK to do (it isn’t), but it's a hill I have no interest in dying on!